### ISSUE TITLE
catch keyboard interrupts during Python execution

### ISSUE THREAD

**Issue by stevengj** [JuliaPy/PyCall.jl#27]
Currently, keyboard interrupts are deferred during execution of Python code, since CPython is not interrupt-safe.  It would be nice to catch these interrupts and set a `KeyboardInterrupt` Python exception so that Python code is halted gracefully.

**Comment by StephenVavasis** [comment]
Currently (Windows 8.1, Julia 0.4.5, PyCall 1.6.1, PyPlot 2.1.1) I see the opposite problem. Once my Julia program starts making PyPlot calls, then ctrl-c is no longer able to interrupt the running Julia program until it reaches the next PyPlot invocation.

**Comment by damiendr** [comment]
Agreed that this would be nice. In my case this mostly bites in combination with IJulia -- if I accidentally ask PyPlot to draw something that takes forever, the only way to interrupt it is to restart the notebook (and lose all the previous results).

**Comment by athre0z** [comment]
This is also my primary reason for annoyance in this otherwise excellent library, also while working in IJulia. If this is reasonably easy to implement, that would be greatly appreciated!

**Comment by Quantisan** [comment]
related discussion over at PythonCall https://github.com/JuliaPy/PythonCall.jl/issues/219
